patroni/api.py

```diff
@@ -239,18 +244,27 @@ def do_POST_restart(self):
 
     @check_auth
     def do_DELETE_restart(self):
-        if self.server.patroni.ha.delete_future_restart():
-            data = "scheduled restart deleted"
-            code = 200
+        if self.server.patroni.ha.is_paused():
+            data = "Can't delete scheduled restart in the paused state"
+            code = 500
         else:
-            data = "no restarts are scheduled"
-            code = 404
+            if self.server.patroni.ha.delete_future_restart():
+                data = "scheduled restart deleted"
+                code = 200
+            else:
+                data = "no restarts are scheduled"
+                code = 404
         self._write_response(code, data)
 
     @check_auth
     def do_POST_reinitialize(self):
         patroni = self.server.patroni
         cluster = patroni.dcs.get_cluster()
+        status_code = 500
+        if self.server.patroni.ha.is_paused():
```

Comment:
We should check that cluster is paused against `cluster` we obtained at line#262, because `Ha` has it's own copy of `cluster` object which might be different.

I think we should create `is_paused` method in the `Cluster` class (patroni/dcs/__init__.py) and `Ha.is_paused` should simply do 'return self.cluster and self.cluster.is_paused()'